File: AdisyonUygulamaSwiftAutoLayout/AdisyonUygulamaSwiftAutoLayout/model/Urun.swift
//
//  Urun.swift
//  AdisyonUygulamaSwiftAutoLayout
//
//  Created by Alperen Saraç on 6.06.2025.
//

import Foundation

struct Urun: Codable {
    let id: Int
    var urunAd: String
    var urunFiyat: Float
    var urunResim: String
    var urunAdet: Int
    var urunKategori: Kategori

    enum CodingKeys: String, CodingKey {
        case id
        case urunAd       = "urun_ad"
        case urunFiyat    = "urun_fiyat"
        case urunResim    = "urun_resim"
        case urunAdet     = "urun_adet"
        case urunKategori
    }
}

File: flutter_adisyon_uygulama/lib/model/urun.dart
import 'kategori.dart';

class Urun {
  final int id;
  String urunAd;
  double urunFiyat;
  String urunResim;
  int urunAdet;
  Kategori urunKategori;

  Urun({
    required this.id,
    required this.urunAd,
    required this.urunFiyat,
    required this.urunResim,
    required this.urunAdet,
    required this.urunKategori,
  });

  factory Urun.fromJson(Map<String, dynamic> json) {
    return Urun(
      id: (json['id'] ?? 0) as int,
      urunAd: (json['urun_ad'] ?? '') as String,
      urunFiyat: ((json['urun_fiyat'] ?? 0) as num).toDouble(),
      urunResim: (json['urun_resim'] ?? '') as String,
      urunAdet: (json['urun_adet'] ?? 0) as int,
      urunKategori: Kategori.fromJson(
        (json['urunKategori'] ?? const <String, dynamic>{}) as Map<String, dynamic>,
      ),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'urun_ad': urunAd,
    'urun_fiyat': urunFiyat,
    'urun_resim': urunResim,
    'urun_adet': urunAdet,
    'urunKategori': urunKategori.toJson(),
  };
}